R/estimateDominance.R

```
@@ -0,0 +1,366 @@
+#' esophagus <- estimateDominance(esophagus, index="does_not_exist")
+#' #Tries to calculate index that is not accepted
+#' # and index that is accepted. Gets an error.
+#' esophagus <- estimateDominance(esophagus, index=c("dmn", "does_not_exist"))
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' #Calculates DBP and Core Abundance indices
+#' esophagus <- estimateDominance(esophagus, index=c("dbp", "core_abundance"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #Shows DBP index
+#' colData(esophagus)$dbp
+#'
+#' #Deletes DBP index
+#' colData(esophagus)$dbp <- NULL
+#' #Shows all indices, DBP is deleted
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Names of columns can be chosen, but the length of arguments must match.
+#' esophagus <- estimateDominance(esophagus,
+#'     index=c("dbp", "core_abundance"),
+#'     name = c("index1", "index2"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #If they do not match, gets an error.
+#' esophagus <- estimateDominance(esophagus,
+#'     index="simpson",
+#'     name = c("index3", "index4"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Calculates all indices
+#' esophagus <- estimateDominance(esophagus)
+#' #Or you can calculate all indices by specify "all"
+#' esophagus <- estimateDominance(esophagus, index="all")
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' @export
+#'
+setGeneric("estimateDominance",signature = c("x"),
+           function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
+                    name = index, ...)
+               standardGeneric("estimateDominance"))
+
+
+#' @rdname estimateDominance
+#' @export
+setMethod("estimateDominance", signature = c(x = "MicrobiomeExperiment"),
+          function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
```

the types of `abund_values` `rank` `as_relative` `aggregate` are not checked. Please apply the "fail early, fail often" maxime

by the way `rank`, if you need to pass it on to the `agglomerateByRank` function needs to be a character. Use `taxonomyRanks(x)[1L]` as the default value

---

Added testing. taxonomyRanks(x)[1L] to default value of rank

---

Changed rank back to numerical values 1 or 2, because it is not a phylogenetic rank. It is used when calculating DBP and DMN. It stores the value 1 or 2 which represents if the most abundant taxon is taken into account (1) or if two most abundant taxa is taken into account (2).

Because rank is used represent phylogenetic rank in many other functions, it might be good idea to consider other name for that to avoid misunderstanding. 

Now, I changed the name of 'rank' to 'ntaxa'.